Content.Server/Chemistry/Components/SolutionPurgeComponent.cs

```diff
@@ -0,0 +1,45 @@
+using Content.Server.Chemistry.EntitySystems;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Chemistry.Components;
+
+/// <summary>
+/// Passively decreases a solution's quantity of reagent(s).
+/// </summary>
+[RegisterComponent]
+[Access(typeof(SolutionPurgeSystem))]
+public sealed class SolutionPurgeComponent : Component
+{
+    /// <summary>
+    /// The name of the solution to detract from.
+    /// </summary>
+    [DataField("solution", required: true), ViewVariables(VVAccess.ReadWrite)]
+    public string Solution = string.Empty;
+
+    /// <summary>
+    /// The reagent(s) to be ignored when purging the solution
+    /// </summary>
+    [DataField("preserve"), ViewVariables(VVAccess.ReadWrite)]
+    public List<string> Preserve = new();
+
+    /// <summary>
+    /// Amount of reagent(s) that are purged
+    /// </summary>
+    [DataField("quantity", required: true), ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 Quantity = default!;
+
+    /// <summary>
+    /// How long it takes to purge once.
+    /// </summary>
+    [DataField("duration"), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan Duration = TimeSpan.FromSeconds(1);
+
+    /// <summary>
+    /// The time when the next purge will occur.
+    /// </summary>
+    [DataField("nextChargeTime", customTypeSerializer: typeof(TimeOffsetSerializer)), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan NextPurgeTime = TimeSpan.FromSeconds(0);
```

Comment:
This code is copy-pasted from solutionregeneration.

Reply:
It's essentially an inverse of it, so it made sense to. It's from my initial "cursed spine" commit

Reply:
could change regeneration to also support taking away the generated solution

Reply:
Whilst putting this together, I did consider whether regeneration and purging should be part of a unified "modify solution on a timer" thing, but felt that they were _different enough_ that separating them was reasonable. Regeneration adds _a specific mixture of reagents_ to a container, while the purger removes some amount of _every reagent_ in a container. The presence of the `Preserve` list just makes things more complex.

I was unable to come up with a decent combination that didn't feel dirty. In the interest of keeping components and systems small and focused, I kept them separate.

Reply:
yeah from reading the system it is reasonable to copy paste it, thought some of the fields could be in a shared component but thats annoying